File: problems/closed-loops/closed-loops.py
loops = {"0": 1, "6": 1, "8": 2, "9": 1}


def closed_loops(num):
	t = 0
	for n in list(str(num)):
		if n in loops:
			t += loops[n]
	return t


assert (closed_loops(2816) == 3)
assert (closed_loops(10000) == 4)
assert (closed_loops(16789) == 4)

File: problems/closed-loops/closed-loops.js
let closedLoops = (number) => {
  let splitNumber = number.toString().split('');
  return splitNumber.reduce((totalLoops, currentNumber) => {
    if(currentNumber === '0' || currentNumber === '6' || currentNumber === '9') {totalLoops++;}
    else if(currentNumber === '8') {totalLoops += 2;}

    return totalLoops;
  }, 0);
};

console.log(`2876 should give us 3 loops`);
console.log(`Program gives us: ${closedLoops(2876)} loops`);
